packages/openapi-2-kong/src/declarative-config/__tests__/plugins.test.js

```
@@ -63,4 +63,244 @@ describe('plugins', () => {
+        const plugin = {
+          config: {},
+        };
+
+        const param = {
+          in: 'query',
+          explode: true,
+          required: false,
+          name: 'some_name',
+          schema: {
+            anyOf: [{ type: 'string' }],
+          },
+          style: 'form',
+        };
+
+        const operation: OA3Operation = {
+          parameters: [param],
+        };
+
+        const generated = generateRequestValidatorPlugin(plugin, operation);
+
+        expect(generated.config).toStrictEqual({
+          version: 'draft4',
+          parameter_schema: [
+            {
+              schema: '{"anyOf":[{"type":"string"}]}',
+              style: param.style,
+              in: param.in,
+              name: param.name,
+              explode: param.explode,
+              required: param.required,
+            },
+          ],
+        });
+      });
+
+      it('should throw error if no operation parameter schema defined', () => {
+        const plugin = {};
+
+        const operation: OA3Operation = {
+          parameters: [
+            {
+              in: 'query',
+              name: 'some_name',
+            },
+          ],
+        };
+
+        expect(() => generateRequestValidatorPlugin(plugin, operation)).toThrowError(
+          "Parameter using 'content' type validation is not supported",
+        );
+      });
+    });
+
+    describe('body_schema and allowed_content_types', () => {
+      it('should not add body_schema or allowed_content_types if no body present', () => {
+        const plugin = {
+          config: {},
+        };
+
+        const operation = {};
+
+        const generated = generateRequestValidatorPlugin(plugin, operation);
+
+        expect(generated.config).toStrictEqual({
+          version: 'draft4',
+          body_schema: '{}',
+        });
+      });
+
+      it('should throw error if no operation request body content defined', () => {
```

not sure what the effect is here.

Consider 200 operations and a request-validator defined.
- 100 POST ones expecting a body (and having it defined)
- 100 GET ones not expecting a body (and hence not defined in the OAS)

Then this would throw errors on the 100 GETs, right? In that case a user cannot specify the root level plugin config, but needs to add 100 specific ones for the POST ones. That will become painful...

---

Gotcha, should it silently continue if thir can't find the appropriate properties? This is me writing a test for existing behaviour but can certainly change it if it's wrong

---

yes. I think that makes sense.

---

Fixed in [`e535aa5` (#3283)](https://github.com/Kong/insomnia/pull/3283/commits/e535aa59f27f60228c96e004952b9220cf6e1263). I also updated the behavior of `parameters_schema` generation to not throw errors.